persistence/src/main/java/org/mskcc/cmo/metadb/persistence/neo4j/MetadbSampleRepository.java

```diff
@@ -57,13 +62,24 @@ MetadbSample findResearchSampleByRequestAndIgoId(@Param("reqId") String reqId,
             @Param("igoId") String igoId);
 
     @Query("MATCH (pa: PatientAlias {namespace: 'cmoId', value: $cmoPatientId})-[:IS_ALIAS]->"
-            + "(p: Patient)-[:HAS_SAMPLE]->(s: Sample)-[:HAS_METADATA]->(sm: SampleMetadata) "
+            + "(p: Patient)-[:HAS_SAMPLE]->(s: Sample {datasource: 'dmp'})-[:HAS_METADATA]->(sm: SampleMetadata) "
+            + "MATCH (r: Request)-[:HAS_SAMPLE]->(s) SET sm.igoRequestId = r.igoRequestId "
+            + "RETURN sm"
+    )
+    List<SampleMetadata> findAllClinicalSampleMetadataListByCmoPatientId(@Param("cmoPatientId") String cmoPatientId);
+
+    @Query("MATCH (pa: PatientAlias {namespace: 'cmoId', value: $cmoPatientId})-[:IS_ALIAS]->"
+            + "(p: Patient)-[:HAS_SAMPLE]->(s: Sample {datasource: 'igo'})-[:HAS_METADATA]->(sm: SampleMetadata) "
             + "MATCH (r: Request)-[:HAS_SAMPLE]->(s) SET sm.igoRequestId = r.igoRequestId "
             + "RETURN sm"
     )
-    List<SampleMetadata> findAllSampleMetadataListByCmoPatientId(@Param("cmoPatientId") String cmoPatientId);
+    List<SampleMetadata> findAllResearchSampleMetadataListByCmoPatientId(@Param("cmoPatientId") String cmoPatientId);
```

Comment:
`findAllResearchSamplesByCmoPatientId`

Reply:
We named it `findAllResearchSampleMetadataListByCmoPatientId` to specify that we're returning sampleMetadata not MetadbSample. Do you want to still change it?